lib/spack/spack/build_environment.py

```diff
@@ -337,6 +338,13 @@ def set_build_environment_variables(pkg, env, dirty):
             if os.path.isdir(pcdir):
                 env.prepend_path('PKG_CONFIG_PATH', pcdir)
 
+    if pkg.architecture.target.module_name:
+        load_module(pkg.architecture.target.module_name)
+    if "cray" in pkg.architecture.platform.name:
```

Comment:
Move to cray.setup_platform_environment()

Also unload `cray-mpich` so we don't accidentally find MPI when we didn't list it as a dependency.